src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -116,21 +116,11 @@ export default class Fms {
         */
         this._previousRouteSegments = [];
 
-        // TODO: Use this!
-        /**
-         * Airport the aircraft arrives to
-         *
-         * @for Fms
-         * @property arrivalAirport
-         * @type {AirportModel}
-         */
-        this.arrivalAirport = null;
-
```

Comment:
Basically to remove all `airport_get()`, and the need for an aircraft or fms to have a `this._airportController` (if they even do). It also serves to separate the arrival/departure fields from the rest of the route, so we can allow/disallow players to change the destination airport of aircraft. Basically the same way we use the `arrivalRunway` and `departureRunway` we would do for the airports (especially useful when this is plural :wink:). And I agree on the circularity; I suppose the runway queue would be an example (though now that is `_id` only :+1:) of how we could end up with a problem there.